### PR TITLE
ENDOC-515 Replace ECR with Local Hub

### PR DESCRIPTION
**Pull request by Lyd1aCla1r3**

### CODE REVIEW

File: vuepress/docs/next/docs/compose/local-hub-overview.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 ---
 sidebarDepth: 2
 ---
-# Entando Component Repository
+# Local Hub
 
 ## Introduction
 
-The Entando Component Repository (ECR) is a repository where 
+The Local Hub is a repository where
```
Comment (nshaw) [inline vuepress/docs/next/docs/compose/local-hub-overview.md:8]: Need to reword this sentence. You can no longer share components from an ECR/Local Hub to other apps or enterprises.

File: vuepress/docs/next/docs/compose/local-hub-overview.md [hunk 2]
```diff
@@ -17,9 +17,9 @@ code that can be used in an Entando widget, page or
 application. Examples of components are widgets, micro frontends,
 content types, labels, plugins, and static resources.
 
-### ECR Bundle
+### Local Hub Bundle
 
-An ECR bundle is a package containing one or more components and a
+A Local Hub bundle is a package containing one or more components and a
 descriptor.yaml file providing information about the bundle. The
 bundle is published in a Git registry and shared with an Entando
```
Comment (nshaw) [inline vuepress/docs/next/docs/compose/local-hub-overview.md:24]: We need to add a TODO to check for language like this. We're switching to v5 which means bundles are published to docker.

File: vuepress/docs/next/docs/compose/local-hub-overview.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ and the K8s service.
 
 ### Example Flow
 
-1.  From the Repository page in the App Builder, the user finds the
+1.  From the Local Hub page in the App Builder, the user finds the
```
Comment (nshaw) [inline vuepress/docs/next/docs/compose/local-hub-overview.md:58]: Hub page

File: vuepress/docs/next/docs/consume/entandoapp-cr.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ spec:
 | :- | :- |
 | `spec.customServerImage`|  Used to deploy the Docker image containing your custom Entando App. Follow these instructions on how to [build your own image](../../tutorials/devops/build-core-image.md). This property and the `spec.standardServerImage` are mutually exclusive.|
 |`spec.dbms` | Allowed values are: MySQL, PostgreSQL (default), Oracle, or embedded. Oracle is only supported as an external database.|
-|`spec.ecrGitSshSecretName`| The configuration used by the Entando Component Repository to download bundles from authenticated Git repositories. It's a Secret containing a private key file named `rsa_id` that matches a public key configured in the authenticated Git repository.|
+|`spec.ecrGitSshSecretName`| The configuration used by the Local Hub to download bundles from authenticated Git repositories. It's a Secret containing a private key file named `rsa_id` that matches a public key configured in the authenticated Git repository.|
```
Comment (nshaw) [inline vuepress/docs/next/docs/consume/entandoapp-cr.md:30]: Local Hub -> Entando Component Manager. In this case, it actually is the CM that does the download.

File: vuepress/docs/next/docs/create/component-gen-overview.md [hunk 5]
```diff
@@ -34,11 +34,11 @@ This section provides an overview of the component generation process. If you wa
 2. Define bundle and deploy assets
    - The Entando blueprint includes scripts and tools to automatically generate a bundle ready for deployment to the ECR. The bundle will include your microservice, generated microfrontends, and any microfrontends that you have created manually under the `ui` folder
    - The generated Spring Boot microservice can be built and deployed to a Docker registry using the provided maven Jib plugin
-   - Generated bundles can be pushed to a git repository of the users choice
+   - Generated bundles can be pushed to a Git repository of the users choice
```
Comment (nshaw) [inline vuepress/docs/next/docs/create/component-gen-overview.md:37]: docker/v5? TODO or fix now???

File: vuepress/docs/next/docs/create/component-gen-overview.md [hunk 6]
```diff
@@ -34,11 +34,11 @@ This section provides an overview of the component generation process. If you wa
...
 
-3. Deploy bundles to [Entando Component Repository (ECR)](../compose/ecr-overview.md)
+3. Deploy bundles to the [Local Hub](../compose/local-hub-overview.md)
    - Once your Docker images are pushed to your registry and your bundle is available in Git you can generate an Entando bundle descriptor and push the bundles to the ECR
-   - The bundles can be versioned and updated using tags in the git repository
+   - The bundles can be versioned and updated using tags in the Git repository
 
 4. Install in Entando Apps
-   - Once the bundles are in the ECR they are available for deployment in any Entando application that has access to the ECR in the cluster
+   - Once the bundles are in the Local Hub they are available for deployment in any Entando application that has access to the Local Hub in the cluster
```
Comment (nshaw) [inline vuepress/docs/next/docs/create/component-gen-overview.md:44]: Drop the "in any Entando application..." language

File: vuepress/docs/next/docs/curate/local-hub-bundle-filters.md [hunk 7]
```diff
@@ -1,12 +1,12 @@
 # Filtering bundles
 
-ECR bundles are filterable by component, by status or by textual research.
+Local Hub bundles are filterable by component, by status or by textual research.
```
Comment (nshaw) [inline vuepress/docs/next/docs/curate/local-hub-bundle-filters.md:3]: Entando Bundles

File: vuepress/docs/next/docs/curate/local-hub-bundle-presentation-config.md [hunk 8]
```diff
@@ -1,8 +1,8 @@
 # How to customize bundle info shown in Entando App Builder
 
-![ECR User interface in Entando App Builder](./img/ecr-page.png)
+![Local Hub user interface in Entando App Builder](./img/local-hub-page.png)
```
Comment (nshaw) [inline vuepress/docs/next/docs/curate/local-hub-bundle-presentation-config.md:3]: Local Hub user interface -> Hub user interface?

File: vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md [hunk 9]
```diff
@@ -1,8 +1,8 @@
 # Bundle versions and updates - FAQ
 
-## Does the ECR support versioning?
-A bundle, as discussed in the [overview](../compose/ecr-overview.md), is a package containing one or more components.
-As in many other packaging systems, the ECR supports bundle versioning, allowing developers to create and release improvements of their package over time.
+## Does the Local Hub support versioning?
+A bundle, as discussed in the [overview](../compose/local-hub-overview.md), is a package containing one or more components.
+As in many other packaging systems, the Local Hub supports bundle versioning, allowing developers to create and release improvements of their package over time.
```
Comment (nshaw) [inline vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md:5]: "the Local Hub " -> "an Entando Bundle"

File: vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md [hunk 10]
```diff
@@ -29,11 +29,11 @@ spec:
 ```
 
 As you can see from the code above, the bundle `cms-quickstart-bundle` has 2 versions available.
-Each version in the `tags` object must correspond a tag in the git repository provided in the `tarball` field.
+Each version in the `tags` object must correspond a tag in the Git repository provided in the `tarball` field.
```
Comment (nshaw) [inline vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md:32]: TODO re v5?

File: vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md [hunk 11]
```diff
@@ -29,11 +29,11 @@ spec:
...
 
 ## What format should I use to version my bundle?
 
-The ECR requires versions to follow the [semantic versioning 2.0.0](https://semver.org/#semantic-versioning-200), with the possibility to prepend a `v` to the version itself. Some valid bundle versions are:
+The Local Hub requires versions to follow the [semantic versioning 2.0.0](https://semver.org/#semantic-versioning-200), with the possibility to prepend a `v` to the version itself. Some valid bundle versions are:
```
Comment (nshaw) [inline vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md:36]: -> "Entanto requires bundles to follow..."

File: vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md [hunk 12]
```diff
@@ -42,8 +42,8 @@ The ECR requires versions to follow the [semantic versioning 2.0.0](https://semv
 
 
 ## As a bundle developer, how should I create a new version of a bundle?
```
Comment (nshaw) [inline vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md:44]: TODO to update this section for docker?

File: vuepress/docs/next/docs/getting-started/concepts-overview.md [hunk 13]
```diff
@@ -43,8 +43,8 @@ The generated skeleton serves as a starting point to help a development team swi
 
 See also: [Entando JHipster Blueprint](../../tutorials/create/ms/generate-microservices-and-micro-frontends.md)
 
-## Entando Component Repository 
-The [Entando Component Repository](../compose/ecr-overview.md) is used to store and retrieve shareable components so they can be used in multiple
+## Local Hub 
+The [Local Hub](../compose/local-hub-overview.md) is used to store and retrieve shareable components so they can be used in multiple
```
Comment (nshaw) [inline vuepress/docs/next/docs/getting-started/concepts-overview.md:47]: Remove "in multiple Entando Applications across an enterprise"

File: vuepress/docs/next/docs/getting-started/concepts-overview.md [hunk 14]
```diff
@@ -112,12 +112,12 @@ are:
 
 #### EntandoDeBundle
 
-An EntandoDeBundle - or Component Repository bundle - is a package with
+An EntandoDeBundle - or Local Hub bundle - is a package with
```
Comment (nshaw) [inline vuepress/docs/next/docs/getting-started/concepts-overview.md:115]: "Local Hub bundle" -> "Entando bundle"

File: vuepress/docs/next/docs/getting-started/ent-bundle.md [hunk 15]
```diff
@@ -12,7 +12,7 @@ This document covers `ent bundle` operations and the steps required to create an
 3. [Run](#run)
 4. [Package](#package)
 5. [Publish](#publish)
-6. [ECR Deploy](#ecr-commands)
+6. [Deploy](#local-hub-commands)
```
Comment (nshaw) [inline vuepress/docs/next/docs/getting-started/ent-bundle.md:15]: This section should remain ecr-commands since the CLI is not changing.

File: vuepress/docs/next/docs/getting-started/ent-bundle.md [hunk 16]
```diff
@@ -98,8 +98,8 @@ See [Build and Publish a Project Bundle](../../tutorials/create/pb/publish-proje
 |`ent bundle publish --org [organization]`|Publish Docker images to the Docker registry, with a specified organization |
 |`ent bundle publish --registry [registry]`| Publish the Docker images to your Docker registry|
 
-## ECR Commands
-Entando provides a series of `ent ecr` commands for managing bundle interactions with [the Entando Component Repository](../../docs/compose/ecr-overview.md) (ECR).
+## Local Hub Commands
```
Comment (nshaw) [inline vuepress/docs/next/docs/getting-started/ent-bundle.md:101]: Keep as "ECR commands"

File: vuepress/docs/next/docs/getting-started/entando-cli.md [hunk 17]
```diff
@@ -53,7 +53,7 @@ Use `ent [command] --help` to see the details for each command.
   - check-env            => Checks the environment for required dependencies and settings
   - diag                 => Runs diagnostics and aggregates the related info in a .tgz file
   - ent                  => Helper for managing the local ent installations
-  - ecr                  => Helper for managing the ECR
+  - ecr                  => Helper for managing the Local Hub
```
Comment (nshaw) [inline vuepress/docs/next/docs/getting-started/entando-cli.md:56]: Revert, it should use the CLI help text.

File: vuepress/docs/next/docs/reference/deployment-structure.md [hunk 18]
```diff
@@ -23,7 +23,7 @@ During installation, an Entando Application needs to create and initialize sever
 * GitHub: <https://github.com/entando-k8s/entando-k8s-dbjob>
 * DockerHub: <https://hub.docker.com/repository/docker/entando/entando-k8s-dbjob>
 
-The screenshot below highlights the init containers for the Entando Application schema creation, DB initialization, and component repository database.
+The screenshot below highlights the init containers for the Entando Application schema creation, DB initialization, and Local Hub database.
```
Comment (nshaw) [inline vuepress/docs/next/docs/reference/deployment-structure.md:26]: Revert, there actually is an ECR database so keeping this as "component repository" is probably safer.

File: vuepress/docs/next/tutorials/curate/local-hub-private-git-repo.md [hunk 19]
```diff
@@ -2,11 +2,11 @@
 # Install Bundles from a Private Git Repository
 
 ## Overview
-The standard deployment of the Entando Component Repository (ECR) assumes that bundles are checked out from public Git repositories. Public repositories do not require user authentication, but if a customer chooses to limit repository access to specific Git accounts, a private key is required. This document details how to allow the ECR to access a private Git repository.
+The standard deployment of the Local Hub assumes that bundles are checked out from public Git repositories. Public repositories do not require user authentication, but if a customer chooses to limit repository access to specific Git accounts, a private key is required. This document details how to allow the Local Hub to access a private Git repository.
```
Comment (nshaw) [inline vuepress/docs/next/tutorials/curate/local-hub-private-git-repo.md:5]: Add a TODO to refine this language for git vs docker

Comment (nshaw) [inline vuepress/docs/next/tutorials/curate/local-hub-private-git-repo.md:5]: Local Hub -> Entando

File: vuepress/docs/next/tutorials/curate/local-hub-private-images.md [hunk 20]
```diff
@@ -5,7 +5,7 @@ sidebarDepth: 2
 # Install Bundle Plugins from a Private Image Repository
 
 ## Overview
-The standard deployment of the Entando Component Repository assumes that plugin images are pulled from public repositories. Public repositores do not require user authentication in order to pull an image. The following tutorial will show you how to add secrets to your Kubernetes environment so you can successfully pull images from private repositories.
+The standard deployment of the Local Hub assumes that plugin images are pulled from public repositories. Public repositores do not require user authentication in order to pull an image. The following tutorial will show you how to add secrets to your Kubernetes environment so you can successfully pull images from private repositories.
```
Comment (nshaw) [inline vuepress/docs/next/tutorials/curate/local-hub-private-images.md:8]: Local Hub -> Entando

File: vuepress/docs/next/tutorials/devops/backing-up-and-restoring-your-environment.md [hunk 21]
```diff
@@ -38,7 +38,7 @@ Restoring a backup requires creating and then deploying a custom image of an Ent
 backup files included. 
 
 ### Build the Custom Image
-1.  Clone the Entando Application repository
+1.  Clone the Entando Application Local Hub
```
Comment (nshaw) [inline vuepress/docs/next/tutorials/devops/backing-up-and-restoring-your-environment.md:41]: Revert, this is unrelated to the hub

File: vuepress/docs/next/tutorials/devops/external-db.md [hunk 22]
```diff
@@ -23,7 +23,7 @@ backup workflow to scale with your application.
 1. Create and install a secret or secrets for your database credentials
 2. Edit the EntandoApp and EntandoKeycloakServer custom resources.
 3. Set the value for `dbms` to `none` in both resources.
-4. Under `environmentVariables`, you will need to provide database connection information for two Entando server databases and one connection for the Entando Component Repository. Example connection information for each of those databases is provided below:
+4. Under `environmentVariables`, you will need to provide database connection information for two Entando server databases and one connection for the Local Hub. Example connection information for each of those databases is provided below:
```
Comment (nshaw) [inline vuepress/docs/next/tutorials/devops/external-db.md:26]: Revert, in this case we need to keep ECR because the properties use it.

**Review by nshaw — CHANGES_REQUESTED** [review]

File: vuepress/docs/next/docs/compose/local-hub-overview.md (hunk 2 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/docs/compose/local-hub-overview.md:24]: @nshaw modified here to briefly touch on both git- and docker-based

File: vuepress/docs/next/docs/create/component-gen-overview.md (hunk 5 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/docs/create/component-gen-overview.md:37]: @nshaw @jyunmitch TODO... all of the details on this page are specific to the jhipster generate ms and mfe as it is now; we'd either want to discuss/link to v1 -> v5 conversion or rewrite for v5 generation (assuming that comes to fruition)

File: vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md (hunk 10 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md:32]: yes, TODO... the example above cites git links so we'll have to revisit this to edit thoroughly/correctly

File: vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md (hunk 12 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/docs/curate/local-hub-bundle-versions-faq.md:44]: yep, TODO

File: vuepress/docs/next/tutorials/curate/local-hub-private-git-repo.md (hunk 19 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/tutorials/curate/local-hub-private-git-repo.md:5]: it looks like the whole tutorial needs to be reworked to include instructions for docker...

File: vuepress/docs/next/docs/compose/local-hub-overview.md [hunk 23]
```diff
@@ -44,9 +40,9 @@ responsible for the low-level communication with the K8s cluster API.
 
 ## Architecture
 
-![ECR Architecture](./img/ecr-architecture.png)
+![Local Hub Architecture](./img/local-hub-architecture.png)
 
-From an architectural point of view, the ECR is composed of: 
+From an architectural point of view, the Local Hub is composed of:
```
Comment (nshaw) [inline vuepress/docs/next/docs/compose/local-hub-overview.md:45]: ECR

File: vuepress/docs/next/docs/compose/local-hub-overview.md [hunk 24]
```diff
@@ -44,9 +40,9 @@ responsible for the low-level communication with the K8s cluster API.
 
 ## Architecture
 
-![ECR Architecture](./img/ecr-architecture.png)
+![Local Hub Architecture](./img/local-hub-architecture.png)
 
-From an architectural point of view, the ECR is composed of: 
+From an architectural point of view, the Local Hub is composed of: 
 1. The `EntandoDeBundles` which contain the metadata associated with a bundle
```
Comment (nshaw) [inline vuepress/docs/next/docs/compose/local-hub-overview.md:46]: Add to #1 or add a new line. "The list of EntandoDeBundles (or Entando Bundles) is displayed as the "Local Hub" in the App Builder."

File: vuepress/docs/next/docs/create/component-gen-overview.md (hunk 5 above)
Comment (nshaw) [reply vuepress/docs/next/docs/create/component-gen-overview.md:37]: @Lyd1aCla1r3 Let's get a new ticket. I think we want to rewrite for v5. I'm requesting updating the blueprint to v5 for 7.1 (https://entando.myjetbrains.com/youtrack/issue/ECS-463/Convert-the-jhipster-blueprint-to-the-new-DockerOCIv5-format)

**Review by nshaw — CHANGES_REQUESTED** [review]

File: vuepress/docs/next/tutorials/curate/local-hub-private-git-repo.md (hunk 19 above)
Comment (nshaw) [reply vuepress/docs/next/tutorials/curate/local-hub-private-git-repo.md]: We don't need to rewrite this tutorial since it is specifically for git-based bundles. I just wanted to refine the language to note that in Entando 7.1 you can also do this with docker.

**Review by nshaw — APPROVED** [review]
